OPHD/UI/Core/Label.cpp

```diff
@@ -22,7 +22,7 @@ Label::Label(std::string newText)
 
 void Label::autoSize()
 {
-	size(textSize() + NAS2D::Vector{FIELD_PADDING * 2, FIELD_PADDING * 2});
+	size(TXT_FONT->size(text()) + NAS2D::Vector{FIELD_PADDING * 2, FIELD_PADDING * 2});
```

Comment:
Here `autoSize` is meant to set the size of the `Control` based on the assigned text, with space for the internal padding. This can be used to ensure text is drawn without any clipping, and so surrounding controls can be placed in accordance with the natural text size.

Reply:
Would you prefer the function name be switched back to autoSize? I had assumed Label's intention was to always resize based on provided text.

Reply:
Ahh, I see. I think the intention I was going for, was to auto size in the constructor when a string is set, but otherwise not override if the size is changed. Leave it to the user to call the `autoSize` method if they want to adjust size after changing the text.

There may be cases where labels should have a fixed size. Perhaps in a grid layout for instance, where cells have a certain maximum width. In that case, the user can call `size` to set a consistent size on a set of `Label` instances.

Reply:
OK, that logic sounds not too difficult to setup.

Reply:
Rough logic anyway.

After writing my last comment, I was wondering about the behaviour when changing the text. Does it call `autoSize` then? Should it? I'm open to discussion on that.

I suppose the main point is you can set `size` after an `autoSize`, so the actual `Label` size might not snugly fit the text. Though unless we clip rendering, perhaps none of this actually matters.

Anyway, hopefully that at least explains why it was the way it was.

Reply:
Yeah, I understand what you are going for. Correct that clipping isn't actually supported. 

I was planning to solve by adding an enum to control resizing.

Really I just wanted to do some general refactor before creating TextControl not a full redesign of Label.